cargo-apk/src/apk.rs

```diff
@@ -47,49 +47,64 @@ impl<'a> ApkBuilder<'a> {
             Artifact::Root(name) => format!("rust.{}", name.replace("-", "_")),
             Artifact::Example(name) => format!("rust.example.{}", name.replace("-", "_")),
         };
-        let package_label = self
-            .manifest
-            .metadata
-            .apk_label
-            .as_deref()
-            .unwrap_or(artifact.name())
-            .to_string();
 
-        let config = Config {
+        // Set default Android manifest values
+        let mut manifest = self.manifest.android_manifest.clone();
+        manifest.package = package_name;
+        manifest.version_name = Some(self.manifest.version.clone());
+        manifest.version_code = Some(VersionCode::from_semver(&self.manifest.version)?.to_code(1));
```

Comment:
Not sure if I like these being unconditionally overriden. Don't think we can/should fail during deserialization, but her should be good to let the user know that their value has no effect:

```suggestion
        assert!(manifest.package.is_empty());
        manifest.package = package_name;
        manifest
            .version_name
            .replace(self.manifest.version.clone())
            .expect("version_name should not be set in TOML");
        manifest
            .version_code
            .replace(VersionCode::from_semver(&self.manifest.version)?.to_code(1))
            .expect("version_code should not be set in TOML");
```

Reply:
Changed it so `cargo-apk` panics when the user specifies these attributes themselves with the suggested error message.

Reply:
Oh right, this should have been an `expect_none` which is not stabilized yet :facepalm: